apps/storefront/pages/_app.tsx

```diff
@@ -30,10 +31,12 @@ function MyApp({ Component, pageProps }: AppPropsWithLayout) {
       <CheckoutProvider>
         <RegionsProvider>
           <SaleorProviderWithChannels>
-            <BaseSeo />
-            <NextNProgress color="#5B68E4" options={{ showSpinner: false }} />
-            {DEMO_MODE && <DemoBanner />}
-            {getLayout(<Component {...pageProps} />)}
+            <SaleorAppSettingsProvider>
```

Comment:
its wrapping entire App but technically it doesnt have to. I can wrap only required component or even remove Provider at all (if header is only place that needs it, I can run fetch() there)